Navigating the submission workflow using the tabs can break

```
Handling error: "Assertion Failed: You modified \"maxStep\" twice on <pass-ember@component:submission-nav::ember1225> in a single render. It was rendered in \"component:submission-nav\" and modified in \"component:workflow-repositories\". This was unreliable and slow in Ember 1.x and is no longer supported. See https://github.com/emberjs/ember.js/issues/13948 for more details."
error-handler.js:16 Error: Assertion Failed: You modified "maxStep" twice on <pass-ember@component:submission-nav::ember1225> in a single render. It was rendered in "component:submission-nav" and modified in "component:workflow-repositories". This was unreliable and slow in Ember 1.x and is no longer supported. See https://github.com/emberjs/ember.js/issues/13948 for more details.
    at new EmberError (error.js:40)
    at Object.assert (index.js:139)
    at TransactionRunner.assertNotRendered (ember-metal.js:876)
    at propertyDidChange (ember-metal.js:1086)
    at ember-metal.js:1158
    at Meta._forEachIn (ember-metal.js:2323)
    at Meta.forEachInDeps (ember-metal.js:2296)
    at iterDeps (ember-metal.js:1146)
    at dependentKeysDidChange (ember-metal.js:1123)
    at propertyDidChange (ember-metal.js:1069)
```

Example:

Navigate through the submission workflow as normal until the "details" page. Click the "repositories" tab to go back to that step. Error occurs.

--- a/app/components/workflow-repositories.js
+++ b/app/components/workflow-repositories.js
@@ -57,16 +57,16 @@ export default Component.extend({
        * as well as any other repositories marked as 'selected'
        */
       if (req) {
-        req.forEach(repoInfo => this.addRepository(repoInfo.repository));
+        req.forEach(repoInfo => this.addRepository(repoInfo.repository, false));
       }
       if (opt) {
         opt.filter(repoInfo => repoInfo.repository.get('_selected'))
-          .forEach(repoInfo => this.addRepository(repoInfo.repository));
+          .forEach(repoInfo => this.addRepository(repoInfo.repository, false));
       }
       if (choice) {
         choice.forEach((group) => {
           group.filter(repoInfo => repoInfo.repository.get('_selected'))
-            .forEach(repoInfo => this.addRepository(repoInfo.repository));
+            .forEach(repoInfo => this.addRepository(repoInfo.repository, false));
         });
       }
     }
@@ -113,23 +113,29 @@ export default Component.extend({
    * Add a repository to the submission only if it is not already included
    *
    * @param {Repository} repository
+   * @param {boolean} setMaxStep should we modify 'maxStep' in the workflow?
    */
-  addRepository(repository) {
+  addRepository(repository, setMaxStep) {
     const subRepos = this.get('submission.repositories');
 
     if (!subRepos.includes(repository)) {
       subRepos.pushObject(repository);
     }
-    this.get('workflow').setMaxStep(4);
+    if (setMaxStep) {
+      this.get('workflow').setMaxStep(4);
+    }
   },
 
   /**
    * Remove a repository from the submission
    *
    * @param {Repository} repository
+   * @param {boolean} setMaxStep should we modify 'maxStep' in the workflow?
    */
-  removeRepository(repository) {
+  removeRepository(repository, setMaxStep) {
     this.get('submission.repositories').removeObject(repository);
-    this.get('workflow').setMaxStep(4);
+    if (setMaxStep) {
+      this.get('workflow').setMaxStep(4);
+    }
   },
 });

--- a/app/components/workflow-repositories.js
+++ b/app/components/workflow-repositories.js
@@ -90,9 +90,9 @@ export default Component.extend({
      */
     toggleRepository(repository, selected, type) {
       if (selected) {
-        this.addRepository(repository);
+        this.addRepository(repository, true);
       } else {
-        this.removeRepository(repository);
+        this.removeRepository(repository, true);
       }
     },
 

